doc/tutorials/introduction/macos_install/macos_install.markdown

```diff
@@ -53,74 +61,121 @@ You can use the latest stable OpenCV version or you can grab the latest snapshot
-    -   set `BUILD_EXAMPLES=ON` to build all examples
+    -   build type: `-DCMAKE_BUILD_TYPE=Release` (or `Debug`).
+    -   include Extra Modules: If you cloned the `opencv_contrib` repository and want to include its modules, set:
+
+        @code{.bash}
+        -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib/modules
+        @endcode
+    -   set `-DBUILD_DOCS=ON` for building documents (doxygen is required)
+    -   set `-DBUILD_EXAMPLES=ON` to build all examples
 
 -#  [optional] Building python. Set the following python parameters:
-    -   `PYTHON3_EXECUTABLE = <path to python>`
-    -   `PYTHON3_INCLUDE_DIR = /usr/include/python<version>`
-    -   `PYTHON3_NUMPY_INCLUDE_DIRS =
-        /usr/lib/python<version>/dist-packages/numpy/core/include/`
+    -   `-DPYTHON3_EXECUTABLE=$(which python3)`
+    -   `-DPYTHON3_INCLUDE_DIR=$(python3 -c "from sysconfig import get_paths as gp; print(gp()['include'])")`
+    -   `-DPYTHON3_NUMPY_INCLUDE_DIRS=$(python3 -c "import numpy; print(numpy.get_include())")`
     @note
-    To specify Python2 versions, you can replace `PYTHON3_` with `PYTHON2_` in the above parameters.
+    Python 2 has reached its end of life. It is recommended to use Python 3 for building Python bindings.
 
 -#  Build. From build directory execute *make*, it is recommended to do this in several threads
 
-    For example
+    For example:
+
+    @code{.bash}
+    make -j$(sysctl -n hw.ncpu) # runs the build using all available CPU cores
+    @endcode
+
+-#  After building, you can install OpenCV system-wide using:
+
+    @code{.bash}
+    sudo make install
+    @endcode
+
+-#  To use OpenCV in your CMake-based projects through `find_package(OpenCV)`, specify the `OpenCV_DIR` variable pointing to the build or install directory.
+
+    For example:
+
+    @code{.bash}
+    cmake -DOpenCV_DIR=~/build_opencv ..
+    @endcode
+
+### Verifying the OpenCV Installation
+
+After building (and optionally installing) OpenCV, you can verify the installation by checking the version using Python:
+
     @code{.bash}
-    make -j7 # runs 7 jobs in parallel
+    python3 -c "import cv2; print(cv2.__version__)"
     @endcode
 
--#  To use OpenCV in your CMake-based projects through `find_package(OpenCV)` specify `OpenCV_DIR=<path_to_build_or_install_directory>` variable.
+This command should output the version of OpenCV you have installed.
 
 @note
 You can also use a package manager like [Homebrew](https://brew.sh/)
 or [pip](https://pip.pypa.io/en/stable/) to install releases of OpenCV only (Not the cutting edge).
+
+- Installing via Homebrew:
+
+    For example:
+
+    @code{.bash}
+    brew install opencv
+    @endcode
+
+- Installing via pip:
+
+    For example:
+
+    @code{.bash}
+    pip install opencv-python
+    @endcode
+
+    @note The `opencv-python` package installed via `pip` does not include non-free algorithms (like SIFT and SURF) and does not include the extra modules from `opencv_contrib`.
```

Comment:
SIFT is included

Reply:
Hmm I think I will write this :
@note The `opencv-python` package installed via `pip` includes the SIFT algorithm starting from OpenCV 4.4. However, it does not include other non-free algorithms (like SURF) or the extra modules from `opencv_contrib`.